### PR TITLE
Task4

### PR DESCRIPTION
**Pull request by MichaelBatsian**
Please review task 4.

### CODE REVIEW

File: Lab4/Task4.1.sql [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	FROM Person.Person AS P
+	INNER JOIN(
+			 HumanResources.Employee AS E
+			 INNER JOIN(
+						HumanResources.EmployeeDepartmentHistory AS DH
+						INNER JOIN HumanResources.Department AS D
+						ON DH.DepartmentID = D.DepartmentID)
+			 ON E.BusinessEntityID = DH.BusinessEntityID)
+	ON P.BusinessEntityID = E.BusinessEntityID 
+	WHERE D.Name != 'Finance' AND P.MiddleName LIKE '[J,L]'
```
[the remark this answers is not in the document]
Comment (MichaelBatsian) [reply Lab4/Task4.1.sql]: fixed

File: Lab4/Task4.2.sql [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+ALTER TABLE  StateProvince ADD CONSTRAINT DF_StateProvince_ModifiedDate 
+	DEFAULT  GETDATE() FOR ModifiedDate 
+
+/*4.2.5
+ Заполните новую таблицу данными из Person.StateProvince. Выберите для вставки только те данные, 
+ где имя штата/государства совпадает с именем страны/региона в таблице CountryRegion
+*/
+
+INSERT INTO StateProvince(
```
[the remark this answers is not in the document]
Comment (MichaelBatsian) [reply Lab4/Task4.2.sql]: It was my mistake.  Value  of the column  "Name"   in the table "CountryRegion" has additional information, more than just country name, that is why it was filtered. I fixed it.

File: Lab4/Task4.1.sql [hunk 3]
```diff
@@ -31,7 +31,7 @@ CONSTRAINT PK_Person_PersonId PRIMARY KEY
 */
 ALTER TABLE Person
 ADD CONSTRAINT CK_Person_MiddleName 
-	CHECK (MiddleName LIKE '[J,L]')
+	CHECK (MiddleName LIKE '[J,L][\.]')
```
Comment (kpavaliayeu) [inline Lab4/Task4.1.sql]: Don't work for me. Did you check this solution?

File: Lab4/Task4.1.sql [hunk 4]
```diff
@@ -78,7 +78,7 @@ INSERT INTO Person(
 						ON DH.DepartmentID = D.DepartmentID)
 			 ON E.BusinessEntityID = DH.BusinessEntityID)
 	ON P.BusinessEntityID = E.BusinessEntityID 
-	WHERE D.Name != 'Finance' AND P.MiddleName LIKE '[J,L]'
+	WHERE  P.MiddleName LIKE '[J,L][\.]'
```
Comment (kpavaliayeu) [inline Lab4/Task4.1.sql]: Don't work for me. Did you check this solution? 
And why did you remove filter for Finance department?

File: Lab4/Task4.2.sql [hunk 5]
```diff
@@ -59,9 +59,11 @@ SELECT
 FROM Person.StateProvince AS SP
 INNER JOIN Person.CountryRegion AS CR 
 ON CR.CountryRegionCode=SP.CountryRegionCode
-WHERE CR.Name =  SP.Name 
+WHERE CR.Name LIKE SP.Name+'%'
```
Comment (kpavaliayeu) [inline Lab4/Task4.2.sql]: I meant that the solution was incorrect. For this case it would work... just before someone didn't update Name with adding prefix. 
Maybe there is no reason to join additional talble ContryRegion... what do you think?

Comment (MichaelBatsian) [reply Lab4/Task4.2.sql]: Yes you're right. I changed this query.
